quic/s2n-quic-h3/src/s2n_quic.rs

```diff
@@ -109,29 +95,60 @@ where
     }
 }
 
+fn convert_connection_error(e: s2n_quic::connection::Error) -> h3::quic::ConnectionErrorIncoming {
+    match e {
+        s2n_quic::connection::Error::Application { error, .. } => {
+            ConnectionErrorIncoming::ApplicationClose {
+                error_code: error.into(),
+            }
+        }
+        s2n_quic::connection::Error::IdleTimerExpired { .. } => ConnectionErrorIncoming::Timeout,
+
+        s2n_quic::connection::Error::Closed { .. } => {
+            // This happens when the connection is closed without an error
+            ConnectionErrorIncoming::InternalError("connection closed".to_string())
+        }
+
+        error => ConnectionErrorIncoming::Undefined(Arc::new(error)),
+    }
+}
+
 impl<B> quic::OpenStreams<B> for Connection
 where
     B: Buf,
 {
     type BidiStream = BidiStream<B>;
     type SendStream = SendStream<B>;
-    type OpenError = ConnectionError;
 
     #[cfg_attr(feature = "tracing", instrument(skip_all, level = "trace"))]
     fn poll_open_bidi(
         &mut self,
         cx: &mut task::Context<'_>,
-    ) -> Poll<Result<Self::BidiStream, Self::OpenError>> {
-        let stream = ready!(self.conn.poll_open_bidirectional_stream(cx))?;
+    ) -> Poll<Result<Self::BidiStream, StreamErrorIncoming>> {
+        let stream = match ready!(self.conn.poll_open_bidirectional_stream(cx)) {
```

Comment:
Nit: This one can also be map_err(), as well as line 144

Reply:
Thanks! I just updated lines with map_ok and map_err.